test/test_jit.py

```diff
@@ -253,5 +255,31 @@ def foo(x) -> Tensor: return x.sum().realize()
     np.testing.assert_allclose(result_2.numpy(), [6], atol=1e-4, rtol=1e-5)
     np.testing.assert_allclose(result_3.numpy(), [6], atol=1e-4, rtol=1e-5)
 
+  @unittest.skipIf(CI and Device.DEFAULT=="METAL", "no ICB in CI, creation of graph fails")
+  def test_jit_batch_split(self):
+    if Device[Device.DEFAULT].graph is None: raise unittest.SkipTest("only test graphs")
+
+    def f(a, b, c, d):
+      x = (a+b).realize()
+      y = (x*c).realize()
+      rn = Tensor.randn(*a.shape)
+      z = (y*rn).realize()
+      w = (z*d)
+      return w.realize()
+
+    a = Tensor.randn(10, 10).realize()
+    b = Tensor.randn(10, 10).realize()
+    c = Tensor.randn(10, 10).realize()
+    d = Tensor.randn(10, 10).realize()
+
+    jf = TinyJit(f)
+    for _ in range(5): jf(a, b, c, d).numpy()
+
+    graph_t = Device[Device.DEFAULT].graph.func if isinstance(Device[Device.DEFAULT].graph, functools.partial) else Device[Device.DEFAULT].graph
+    assert len(jf.jit_cache) == 3, "2 graphs and rand op"
```

Comment:
Whenever you write tests, I want you to think about how many refactors they will survive.

What's going to happen when we switch rand to threefry?

Many of your tests just end up getting deleted because they are fragile.